Client/Frontend/Browser/TabManager.swift

```diff
@@ -403,83 +411,47 @@ class TabManager: NSObject {
             return
         }
 
-        if tab.isPrivate {
-            removeAllBrowsingDataForTab(tab)
-        }
-
-        let oldSelectedTab = selectedTab
-
         if notify {
             delegates.forEach { $0.get()?.tabManager(self, willRemoveTab: tab) }
         }
 
-        // The index of the tab in its respective tab grouping. Used to figure out which tab is next
-        var tabIndex: Int = -1
-        if let oldTab = oldSelectedTab {
-            tabIndex = (tab.isPrivate ? privateTabs.index(of: oldTab) : normalTabs.index(of: oldTab)) ?? -1
-        }
-
         let prevCount = count
         tabs.remove(at: removalIndex)
-
-        let viableTabs: [Tab] = tab.isPrivate ? privateTabs : normalTabs
-
-        // Let's select the tab to be selected next.
-        if let oldTab = oldSelectedTab, tab !== oldTab {
-            // If it wasn't the selected tab we removed, then keep it like that.
-            // It might have changed index, so we look it up again.
-            _selectedIndex = tabs.index(of: oldTab) ?? -1
-        } else if let parentTab = tab.parent,
-            let newTab = viableTabs.reduce(viableTabs.first, { currentBestTab, tab2 in
-            if let tab1 = currentBestTab, let time1 = tab1.lastExecutedTime {
-                if let time2 = tab2.lastExecutedTime {
-                    return time1 <= time2 ? tab2 : tab1
-                }
-                return tab1
-            } else {
-                return tab2
-            }
-        }), parentTab == newTab, tab !== newTab, newTab.lastExecutedTime != nil {
-            // We select the most recently visited tab, only if it is also the parent tab of the closed tab.
-            _selectedIndex = tabs.index(of: newTab) ?? -1
-        } else {
-            // By now, we've just removed the selected one, and no previously loaded
-            // tabs. So let's load the final one in the tab tray.
-            if tabIndex == viableTabs.count {
-                tabIndex -= 1
-            }
-
-            if let currentTab = viableTabs[safe: tabIndex] {
-                _selectedIndex = tabs.index(of: currentTab) ?? -1
-            } else {
-                _selectedIndex = -1
-            }
-        }
-
         assert(count == prevCount - 1, "Make sure the tab count was actually removed")
```

Comment:
I wonder if this is necessary? Do we really need to check tabs.remove(at: removalIndex) ?